docs/upgrade.html

```diff
@@ -67,6 +67,9 @@ <h5><a id="upgrade_320_notable" href="#upgrade_320_notable">Notable changes in 3
             which meant that idempotence remained disabled unless the user had explicitly set <code>enable.idempotence</code> to true
             (See <a href="https://issues.apache.org/jira/browse/KAFKA-13598">KAFKA-13598</a>for more details).
             This issue was fixed and the default is properly applied in 3.0.1, 3.1.1, and 3.2.0.</li>
+        <ul>
+            <li>When the broker version is lower than 2.8.0, and the client version is 3.0.1, 3.1.1, and later, the IDEMPOTENT_WRITE permission is required to produce data</li>
```

Comment:
I would say something like: Please read the relevant KIP section for the compatibility implications - a noteworthy item worth highlighting is...

Also, we should use the same approach for this and other lines. It's unclear to me why we added a `ul` here, but not in the other case. I'd probably include this paragraph in the previous <li>

Reply:
Sorry, I was unclear here. I clarified in https://github.com/apache/kafka/pull/12260#issuecomment-1213588315